libs/SalesforceSDKCore/SalesforceSDKCore/Classes/NativeLoginManagerInternal.swift

```diff
@@ -215,7 +211,303 @@ public class NativeLoginManagerInternal: NSObject, NativeLoginManager {
         return urlSafeBase64Encode(data: hash.dataRepresentation)
     }
     
-    // MARK: Headless, Password-Less Login Via One-Time-Passcode
+    // MARK: Salesforce Identity API Headless Registration Flow
+    
+    @objc public func startRegistration(
+        email: String,
+        firstName: String,
+        lastName: String,
+        username: String,
+        newPassword: String,
+        reCaptchaToken: String,
+        otpVerificationMethod: OtpVerificationMethod
+    ) async -> StartRegistrationResult {
+        
+        // Validate parameters.
+        let trimmedEmail = email.trimmingCharacters(in: .whitespacesAndNewlines)
+        let trimmedUsername = username.trimmingCharacters(in: .whitespacesAndNewlines)
+        let trimmedPassword = newPassword.trimmingCharacters(in: .whitespacesAndNewlines)
+        if let it = mapInvalidEmailAddressToResult(trimmedEmail) { return StartRegistrationResult(nativeLoginResult: it) }
+        if let it = mapInvalidUsernameToResult(trimmedUsername) { return StartRegistrationResult(nativeLoginResult: it) }
+        if let it = mapInvalidPasswordToResult(trimmedPassword) { return StartRegistrationResult(nativeLoginResult: it) }
+        let reCaptchaParameterGenerationResult = generateReCaptchaParameters(reCaptchaToken: reCaptchaToken)
+        
+        // Determine the OTP verification method.
+        let otpVerificationMethodString = generateVerificationTypeHeaderValue(
+            otpVerificationMethod: otpVerificationMethod)
+        
+        // Generate the start registration request body.
+        guard let startRegistrationRequestBodyString = {
+            do { return String(
+                data: try JSONEncoder().encode(
+                    StartRegistrationRequestBody(
+                        recaptcha: reCaptchaParameterGenerationResult.nonEnterpriseReCaptchaToken,
+                        recaptchaevent: reCaptchaParameterGenerationResult.enterpriseReCaptchaEvent,
+                        userData: UserData(
+                            email: email,
+                            username: username,
+                            password: newPassword,
+                            firstName: firstName,
+                            lastName: lastName),
+                        otpVerificationMethod: otpVerificationMethodString)
+                ),
+                encoding: .utf8)!
```

Comment:
Why is this force unwrapped? Is there any risk of hitting it?

Reply:
That is totally left-over from earlier code.  Great catch!

Reply:
Can you decode JSON without the force unwrap?

Reply:
I just verified the unwrap isn't needed 👍🏻